boards/nordic/nrf54h20dk/Kconfig.nrf54h20dk

```diff
@@ -3,12 +3,14 @@
 
 config BOARD_NRF54H20DK
 	select SOC_NRF54H20_CPUAPP if (BOARD_NRF54H20DK_NRF54H20_CPUAPP || \
-				       BOARD_NRF54H20DK_NRF54H20_CPUAPP_IRON)
+				       BOARD_NRF54H20DK_NRF54H20_CPUAPP_IRON || \
+					   BOARD_NRF54H20DK_NRF54H20_CPUAPP_IRON_MCUBOOT_B0)
 	select SOC_NRF54H20_CPURAD if (BOARD_NRF54H20DK_NRF54H20_CPURAD || \
 				       BOARD_NRF54H20DK_NRF54H20_CPURAD_IRON)
 	select SOC_NRF54H20_CPUPPR if (BOARD_NRF54H20DK_NRF54H20_CPUPPR || \
 				       BOARD_NRF54H20DK_NRF54H20_CPUPPR_XIP)
 	select SOC_NRF54H20_CPUFLPR if (BOARD_NRF54H20DK_NRF54H20_CPUFLPR || \
 				        BOARD_NRF54H20DK_NRF54H20_CPUFLPR_XIP)
 	select SOC_NRF54H20_IRON if (BOARD_NRF54H20DK_NRF54H20_CPUAPP_IRON || \
-				       BOARD_NRF54H20DK_NRF54H20_CPURAD_IRON)
+				       BOARD_NRF54H20DK_NRF54H20_CPURAD_IRON || \
+					   BOARD_NRF54H20DK_NRF54H20_CPUAPP_IRON_MCUBOOT_B0)
```

Comment:
I think we will need the CPURAD board soon too, but we may handle it in a separate PR.